Kernel/System/DynamicField/Driver/Set.pm

```diff
@@ -130,37 +200,33 @@ sub ValueSet {
     my $DynamicFieldObject = $Kernel::OM->Get('Kernel::System::DynamicField');
     my $BackendObject      = $Kernel::OM->Get('Kernel::System::DynamicField::Backend');
 
-    # loop over the dynamic fields that are included in the Set
-    for my $i ( 0 .. $#{ $Param{DynamicFieldConfig}{Config}{Include} } ) {
-
-        # historical Definitions will be provided by ITSMConfigItems
-        my $Include      = $Param{DynamicFieldConfig}{Config}{Include}[$i];
-        my $DynamicField = $Include->{Definition} // $DynamicFieldObject->DynamicFieldGet(
-            Name => $Include->{DF},
-        );
-
-        if ( !$DynamicField ) {
-            $Kernel::OM->Get('Kernel::System::Log')->Log(
-                Priority => 'error',
-                Message  => "$Param{DynamicFieldConfig}{Name} configured erroneously. 'DF: DynamicFieldName' missing or wrong.",
-            );
-
-            return;
-        }
-
-        # The values for an included dynamic field are the values from the respective column
-        my @FieldValue = map { $_->[$i] } @SetValue;
-
-        if (
-            !$BackendObject->ValueSet(
-                %Param,
-                DynamicFieldConfig => $DynamicField,
-                Value              => \@FieldValue,
-                Set                => 1,
-            )
+    my $Include = $Param{DynamicFieldConfig}{Config}{Include};
+    my $DynamicFieldName = $Param{DynamicFieldConfig}{Name};
+    my $DynamicFields = $GetIncludedDynamicFields->($Include, $DynamicFieldObject, $DynamicFieldName);
+
+    if (!$DynamicFields) {
+        
+        $Success = 0;
+    } else {
```

Comment:
Please `return;` instead of `return 0;` or `return undef;` or so, in case of errors. (With the exception of something like `return { Success => 0, Error => 'Did not work' };` where you explicitely want to return more information even in case of an error.)

Here the usual way would be:
```
return if !$DynamicFields;

for my $i ( 0 .. $#{ $DynamicFields } ) {
    ...
```

(One could even argue whether a ValueSet() for a set which has no dynamic fields in it should just `return 1;`, actually, but it's a more theoretical question, I guess...)